tools/c7n_gcp/c7n_gcp/resources/logging.py

```diff
@@ -18,20 +18,134 @@
+            return client.execute_query('get', {
                 'sinkName': 'projects/{project_id}/sinks/{name}'.format(
                     **resource_info)})
+
+
+@resources.register('log-project-sink')
+class LoggingProjectSink(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'logging'
+        version = 'v2'
+        component = 'projects.sinks'
+        enum_spec = ('list', 'sinks[]', None)
+        scope_key = 'parent'
+        scope_template = 'projects/{}'
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_query('get', {
+                'sinkName': 'projects/{project_id}/sinks/{name}'.format(
+                    **resource_info)})
+
+
+@resources.register('log-project-metric')
+class LoggingProjectMetric(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'logging'
+        version = 'v2'
+        component = 'projects.metrics'
+        enum_spec = ('list', 'metrics[]', None)
+        scope_key = 'parent'
+        scope_template = 'projects/{}'
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_query('get', {
+                'metricName': 'projects/{project_id}/metrics/{name}'.format(
+                    **resource_info)})
+
+
+@resources.register('log-project-log')
+class LoggingProject(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'logging'
+        version = 'v2'
+        component = 'projects.logs'
+        enum_spec = ('list', 'logNames[]', None)
+        scope_key = 'parent'
+        scope_template = 'projects/{}'
+        id = 'name'
+
+
+@resources.register('log-project-exclusion')
+class LoggingProjectExclusion(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'logging'
+        version = 'v2'
+        component = 'projects.exclusions'
+        enum_spec = ('list', 'exclusions[]', None)
+        scope_key = 'parent'
+        scope_template = 'projects/{}'
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_query('get', {
+                'name': 'projects/{project_id}/exclusions/{exclusion_id}'.format(
+                    **resource_info)})
+
+
+@resources.register('log-exclusion')
+class LoggingExclusion(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'logging'
+        version = 'v2'
+        component = 'exclusions'
+        enum_spec = ('list', 'exclusions[]', None)
+        scope_key = 'parent'
+        scope_template = 'projects/{}'
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_query('get', {
+                'name': '{type}/{project_id}/exclusions/{exclusion_id}'.format(
+                    **resource_info)})
+
+
+@resources.register('log-log')
```

Comment:
besides the akward naming, how is this different the the log-project-log?

Reply:
Removed log resource, because it duplicate log-project-log.

Logs for billingAccount, organizations and folders will be implemented after POCs applying.